Source/Layout/ASLayout+IGListKit.mm

```diff
@@ -0,0 +1,35 @@
+//
+//  ASLayout+IGListKit.mm
+//  Texture
+//
+//  Copyright (c) 2018-present, Pinterest, Inc.  All rights reserved.
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+#import <AsyncDisplayKit/ASAvailability.h>
+#if AS_IG_LIST_KIT
+#import "ASLayout+IGListKit.h"
+#import <AsyncDisplayKit/ASLayout.h>
+#import <IGListKit/IGListKit.h>
+#import <AsyncDisplayKit/ASEqualityHelpers.h>
+
+@implementation ASLayout(IGListKit)
+- (nonnull id <NSObject>)diffIdentifier
+{
+  return @([self.layoutElement hash]);
+}
+
+- (BOOL)isEqualToDiffableObject:(nullable id <IGListDiffable>)other
+{
+  if (other == self) return YES;
+
+  ASLayout *otherLayout = ASDynamicCast(other, ASLayout);
+  if (!otherLayout) return NO;
+
+  return [otherLayout.layoutElement isEqual:self.layoutElement];
```

Comment:
Nit: remove nullability annotations from implementation file to be consistent with the rest of the framework style.

If it's easy to do, import the header that exposes the `_layoutElement` instance variable and use that directly to avoid extra retain/release on these elements. If not, it's not a blocker for this diff.